restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java

```diff
@@ -2595,6 +2595,13 @@ private static void validateFinderMethod(final ResourceMethodDescriptor finderMe
     Method method = finderMethodDescriptor.getMethod();
     Class<?> valueClass = resourceModel.getValueClass();
 
+    if (SingleUnstructuredDataResource.class.isAssignableFrom(resourceModel.getResourceClass())
```

Comment:
Although this is alright, an alternative would be to leverage the logic in `ResourceModel#getResourceEntityType` to do this check without it being class-specific. You can avoid logic duplication this way.

Reply:
cool, makes me know one more thing